include/mesh/boundary_info.h

```diff
@@ -550,6 +557,20 @@ class BoundaryInfo : public ParallelObject
    */
   void build_shellface_boundary_ids(std::vector<boundary_id_type> & b_ids) const;
 
+  /**
+  * Update parent's boundary id list so that this information is consistent with
+  * its children
+  *
+  * This is useful when `_children_on_boundary = true`, and is used when the
+  * element is about to get coarsened i.e., in MeshRefinement::_coarsen_elements()
+  *
+  * Specifically, when we coarsen an element whose children have different boundary ids.
+  * In such scenarios, the parent will inherit the children's boundaries if at
+  * least two of them own a boundary while sharing the side of the parent.
```

Comment:
Surely "two" should be dimension-dependent?  If we're in 1D, for example, and there's only 1 child on a side to begin with, then we still definitely want to copy that id back to the parent!  In 2D "2" is reasonable but then the equivalent criterion in 3D would be "4"; alternatively "2" would also be reasonable in 3D but in that case the equivalent criterion in 2D would be "1".

Reply:
Oh yes, this came from my 2D mindset. I'll do "if 50%+" of the children on that side have it.